libgalois/include/katana/LC_CSR_CSC_Labeled_Graph.h

```diff
@@ -422,52 +329,34 @@ class LC_CSR_CSC_Labeled_Graph
       return std::nullopt;
     }
 
-    auto first = FindFirstOrLastEdgeWithLabel<
-        false, FindEdgeWithLabelOptions::FindFirst>(node, key, data);
-    if (!first) {
+    edge_iterator begin = raw_begin(node, data);
+    edge_iterator end = raw_end(node, data);
+    auto range = internal::make_no_deref_range(begin, end);
+    EdgeIteratorComparator<false> comp(this);
+    auto [first_it, last_it] =
+        std::equal_range(range.begin(), range.end(), key, comp);
+    if (*first_it == end || this->getEdgeDst(*first_it) != key) {
       return std::nullopt;
     }
-
-    auto last = FindFirstOrLastEdge<false, FindEdgeWithLabelOptions::FindLast>(
-        std::get<1>(*first), std::get<2>(*first), key);
-    KATANA_LOG_DEBUG_ASSERT(last);
-
-    return internal::make_no_deref_range(
-        std::get<0>(*first), std::get<0>(*last) + 1);
+    KATANA_LOG_DEBUG_ASSERT(this->getEdgeDst(*last_it - 1) == key);
+    return internal::make_no_deref_range(*first_it, *last_it);
   }
 
-  /**
-   * Returns an edge iterator to an edge with some node and key by
-   * searching for the key via the node's outgoing or incoming edges.
-   * If not found, returns nothing.
-   */
   template <bool in_edges>
-  std::optional<edge_iterator> FindEdge(GraphNode node, GraphNode key) const {
+  bool IsConnectedWithEdgeLabelDirected(
+      GraphNode node, GraphNode key, const EdgeTy& data) const {
     // trivial check; can't be connected if degree is 0
-    if (in_edges) {
-      if (in_degrees_[node] == 0 || degrees_[key] == 0) {
-        return std::nullopt;
-      }
-    } else {
-      if (degrees_[node] == 0 || in_degrees_[key] == 0) {
-        return std::nullopt;
-      }
+    if (degrees_[node] == 0 || in_degrees_[key] == 0) {
```

Comment:
Why this change? This is wrong if `in_edges` is true.

Reply:
whoops! You're right, I'll add it back